demo_nodes_cpp/src/topics/talker.cpp

```diff
@@ -56,9 +56,8 @@ class Talker : public rclcpp::Node
       };
 
     // Create a publisher with a custom Quality of Service profile.
-    rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
-    custom_qos_profile.depth = 7;
-    pub_ = this->create_publisher<std_msgs::msg::String>(topic_name, custom_qos_profile);
+    rclcpp::QoS qos(rclcpp::KeepLast(7));
+    pub_ = this->create_publisher<std_msgs::msg::String>(topic_name, qos);
```

Comment:
I intentionally left this one like this, because the comment is that we're demonstrating how to create a custom QoS (changing the depth to 7 from the default of 10 is kind of meaningless other than just for demonstration), so I decided in this case only to leave the explicit initialization of the QoS and show how they could create one and customize it before passing it.